utils/resb/src/binary/deserializer.rs

```diff
@@ -0,0 +1,1117 @@
+        self.deserialize_seq(visitor)
+    }
+
+    fn deserialize_tuple_struct<V>(
+        self,
+        _name: &'static str,
+        _len: usize,
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        // We copy `serde_json` in handling tuples as sequences.
+        self.deserialize_seq(visitor)
+    }
+
+    fn deserialize_map<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        let descriptor = self.get_next_resource_descriptor()?;
+        match descriptor.resource_type() {
+            ResourceReprType::Table => {
+                if descriptor.is_empty() {
+                    // Handle empty descriptors per-type so we don't miss a type
+                    // mismatch.
+                    return visitor.visit_map(EmptyMapAccess);
+                }
+
+                let input =
+                    get_subslice(self.body, get_32_bit_offset(descriptor.offset() as usize)..)?;
+                let (length, keys) = read_u16(input)?;
+
+                // Most values in the file are 32-bit aligned, so sequences of
+                // 16-bit values may be padded.
+                let length_with_padding = (length + ((length + 1) % 2)) as usize;
+
+                let values_offset = get_16_bit_offset(length_with_padding);
+                let values = get_subslice(keys, values_offset..)?;
+
+                visitor.visit_map(TableMapAccess {
+                    de: self,
+                    keys,
+                    values,
+                    remaining: length as usize,
+                })
+            }
+            ResourceReprType::_Table32 => todo!(),
+            ResourceReprType::Table16 => todo!(),
+            _ => Err(Error::new("Unexpected resource type where table expected.")),
+        }
+    }
+
+    fn deserialize_struct<V>(
+        self,
+        _name: &'static str,
+        _fields: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        self.deserialize_map(visitor)
+    }
+
+    fn deserialize_enum<V>(
+        self,
+        _name: &'static str,
+        _variants: &'static [&'static str],
+        _visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        // Resource bundles have no concept of an enum and it's unclear how to
+        // handle untagged heterogeneous values.
+        todo!()
+    }
+
+    fn deserialize_identifier<V>(self, _visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        unimplemented!()
+    }
+
+    fn deserialize_ignored_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        let (_, input) = read_u32(self.input)?;
+        self.input = input;
+
+        visitor.visit_none()
+    }
+
+    fn is_human_readable(&self) -> bool {
```

Comment:
Nit: Mark this as `#[inline]` since this is often use for DCE

Reply:
Addressed.